runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathIndicatorPlugin.java

```diff
@@ -78,12 +78,17 @@
 	@Inject
 	private InfoBoxManager infoBoxManager;
 
+	@Inject
+	private DeathIndicatorOverlay overlay;
+
 	private Timer deathTimer;
 
 	private WorldPoint lastDeath;
 	private Instant lastDeathTime;
 	private int lastDeathWorld;
 
+	private boolean highlightTile = false;
```

Comment:
This makes no sense. Just check isDead/the config option.

Reply:
Good call, I went ahead and fixed this so that the tile is only rendered if the player hasDied, config.showDeathTileHighlight, and onDeathWorld. I also removed all instance of that highlightTile boolean since it was pointless.

To further clarify:
```java
// Check that the player wants tile highlight, has died, and is on the death world
if (!config.showDeathTileHighlight()
	|| !plugin.hasDied()
	|| !plugin.onDeathWorld())
{
	return null;
}

// Render tile highlight if player wants it, has died, and is on the correct world.
```

 Let me know if there is anything else you'd like changed!